lib/_emerge/Package.py

```diff
@@ -584,33 +583,27 @@ def __str__(self):
 
         build_id_str = ""
         if isinstance(self.cpv.build_id, int) and self.cpv.build_id > 0:
-            build_id_str = "-%s" % self.cpv.build_id
-
-        s = "(%s, %s" % (
-            portage.output.colorize(
-                cpv_color,
-                self.cpv
-                + build_id_str
-                + _slot_separator
-                + self.slot
-                + "/"
-                + self.sub_slot
-                + _repo_separator
-                + self.repo,
+            build_id_str = f"-{self.cpv.build_id}"
+
+        output = portage.output.colorize(
+            cpv_color,
+            (
+                f"{self.cpv}{build_id_str}{_slot_separator}{self.slot}/"
+                f"{self.sub_slot}{_repo_separator}{self.repo}"
             ),
-            self.type_name,
         )
+        s = f"({output}, {self.type_name}"
 
         if self.type_name == "installed":
             if self.root_config.settings["ROOT"] != "/":
-                s += " in '%s'" % self.root_config.settings["ROOT"]
+                s = f"{s} in '{self.root_config.settings['ROOT']}'"
             if self.operation == "uninstall":
-                s += " scheduled for uninstall"
+                s = f"{s} scheduled for uninstall"
         else:
             if self.operation == "merge":
-                s += " scheduled for merge"
+                s = f"{s} scheduled for merge"
                 if self.root_config.settings["ROOT"] != "/":
-                    s += " to '%s'" % self.root_config.settings["ROOT"]
+                    s = f"{s} to '{self.root_config.settings['ROOT']}'"
```

Comment:
Instead of `s = f"{s} ...` you can do `s += f" ..."`
No idea if this is better or not?

Reply:
The point of doing the former is to avoid doing the latter. Formatting strings together is always faster than concatenating.

Reply:
OK, didn't know that - thanks @kenrap for this new information!

Reply:
@arthurzam

Looks like I was wrong. Formatting strings together is slower than concatenating in most cases. Honestly, I'm as surprised as what this stackoverflow user had wrote: https://stackoverflow.com/a/59183178